### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by viru2001**
- Added Login , logout and Signup functionalities
- Added Homepage screen

### CODE REVIEW

File: src/frontend/components/AppBar/AppBar.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                sx={{
+                  backgroundColor: "secondary.main",
+                  color: "#363636",
+                  ":hover": { backgroundColor: "secondary.light" },
+                }}
+                onClick={() => logoutHandler(setAuth)}
+              >
+                Logout
+              </Button>
+              {/* <Typography
```
Comment (irajbhinde) [inline src/frontend/components/AppBar/AppBar.jsx]: Since the auth is implemented, why not uncomment this code

File: public/index.html [hunk 2]
```diff
@@ -24,7 +24,16 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+     <!-- Material Icons -->
```
Comment (irajbhinde) [inline public/index.html]: Remove the commented code

File: src/frontend/components/VideoList/VideoList.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const VideoList = () => {
+  return (
+    <Box
+      component="main"
+      sx={{ flexGrow: 1, p: 3, mt: 8, backgroundColor: "primary.faint" }}
+    >
+      <Grid
+        container
+        spacing={{ xs: 2, md: 3 }}
+        // columns={{ xs: 4, sm: 8, md: 12 }}
```
Comment (irajbhinde) [inline src/frontend/components/VideoList/VideoList.jsx:13]: Remove the commented code

File: src/frontend/pages/Login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+              Login as Guest
+            </Button>
+            <Box
+              sx={{
+                display: "flex",
+                justifyContent: "center",
+              }}
+            >
+              <Link to="/signup" variant="body2" className="link">
+                {"Don't have an account? Sign Up"}
```
Comment (irajbhinde) [inline src/frontend/pages/Login/Login.jsx:156]: You can add a little padding-bottom here for better UX

File: src/frontend/pages/Signup/Signup.jsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+              fullWidth
+              variant="contained"
+              sx={{ mt: 3, mb: 2 }}
+              color="secondary"
+            >
+              Sign Up
+            </Button>
+            <Box sx={{ display: "flex", justifyContent: "center" }}>
+              <Link to="/login" variant="body2" className="link">
+                Already have an account? Log in
```
Comment (irajbhinde) [inline src/frontend/pages/Signup/Signup.jsx:201]: Same here

**Review by irajbhinde — COMMENTED** [review]
As per P0 user stories in LMS - We have to create browse by categories as well. You can create a set of chips just above your videos container to browse by category

Other than that, everything looks real clean

File: src/frontend/components/AppBar/AppBar.jsx (hunk 1 above)
Comment (viru2001) [reply src/frontend/components/AppBar/AppBar.jsx]: yes I will uncomment it

File: src/frontend/components/VideoList/VideoList.jsx (hunk 3 above)
Comment (viru2001) [reply src/frontend/components/VideoList/VideoList.jsx:13]: this attribute i kept to make grid responsive . will utilize it in future

File: src/frontend/components/AppBar/AppBar.jsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  const logoutHandler = setAuth => {
+    localStorage.removeItem("AUTH_TOKEN");
+    localStorage.removeItem("username");
+    setAuth(auth => ({
+      ...auth,
+      status: false,
+      token: null,
+      username: "",
+    }));
+  };
```
Comment (vishalg8454) [inline src/frontend/components/AppBar/AppBar.jsx:45]: I would suggest moving the logout code from component to your context.

File: src/frontend/components/KebabMenu/KebabMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import { Icon } from "@mui/material";
+import { PlaylistDialog } from "../PlaylistDialog/PlaylistDialog";
+
+const ITEM_HEIGHT = 48;
+
+function KebabMenu() {
+  const [anchorEl, setAnchorEl] = React.useState(null);
```
Comment (vishalg8454) [inline src/frontend/components/KebabMenu/KebabMenu.jsx:12]: why not import `useState` directly from 'react'?

**Review by vishalg8454 — APPROVED** [review]
Clean UI and color theme. Left a couple of suggestions.
Also check the positioning of three dot menu in video card.
image

**Pull request merged**